plugins/module_utils/wdc_redfish_utils.py

```diff
@@ -0,0 +1,542 @@
+# -*- coding: utf-8 -*-
+
+# Copyright (c) 2022 Western Digital Corporation
+# GNU General Public License v3.0+ (see LICENSE or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+import datetime
+import re
+import time
+import tarfile
+import socket
+
+from ansible.module_utils.urls import fetch_file, open_url
+from ansible_collections.community.general.plugins.module_utils.redfish_utils import RedfishUtils
+
+try:
+    # Python3
+    from urllib.parse import urlparse, urlunparse
+except ImportError:
+    # Python2
+    from urlparse import urlparse, urlunparse
+
+try:
+    import ipaddress
+    IPADDRESS_AVAILABLE = True
+except ImportError:
+    IPADDRESS_AVAILABLE = False
+
+
+try:
+    from dns import resolver
+    from dns.resolver import NXDOMAIN
+    DNS_AVAILABLE = True
+except ImportError:
+    DNS_AVAILABLE = False
+
+
+class DnsCacheBypass:
```

Comment:
Is this something that's really necessary? I'd hate to have this sprinkled before every request made to a Redfish service. If IP addresses are changing on devices frequently, shouldn't they have a short life span in the DNS cache?

Reply:
Unfortunately it's necessary when the controller is reset (which happens as part of firmware update).  If we don't have this, subsequent Redfish commands will fail until the cache entry expires or the user manually flushes the cache.  It is a known issue with this particular storage controller.